dataframe/dataframe.go

```diff
@@ -434,6 +434,62 @@ func (df DataFrame) GroupBy(colnames ...string) *Groups {
 	return groups
 }
 
+type ColumnValuesPerSubset struct {
+	Values      series.Series
+	SubsetIndex series.Indexes
+}
+
+type ColumnUpdateRule struct {
```

Comment:
I can't find better name for `ColumnUpdateRule`. Probably, it could be called `ColumnUpdate` ?